src/common/freeze/deepfreeze.ts

```diff
@@ -0,0 +1,21 @@
+
```

Comment:
There is a property `Object.freeze` see [link](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Object/freeze) that we can use to freeze the object and then use chai method `to.be.frozen` to expect the result. Example here:

````javascript
 it('should return original state when passing not expected action type', () => {
    //Arrange
    const originalState = new TrainingState();
    const action = {
      type: 'NOT_EXPECTED_ACTION'
    };

    //Act
    Object.freeze(originalState);
    const nextState = trainingReducer(originalState, action);

    //Assert
    expect(nextState).not.to.be.undefined;
    expect(nextState).to.equal(originalState);
    expect(originalState).to.be.frozen;
  });
```